### ISSUE TITLE
Key with spaces does not always work in v21

### ISSUE THREAD

**Issue by pasieronen** [i18next/i18next#1661]
## 🐛 Bug Report

It looks like detecting natural language keys (2cb25e4b) changed behavior for keys containing spaces.

## To Reproduce

```js
const i18next = require('i18next')

i18next.init({
  lng: 'en',
  resources: {
    en: {
      translation: {
        errorCodes: {
          'UNAUTHORIZED': 'Unauthorized',
          'BAD REQUEST': 'Bad request'
        }
      }
    }
  }
})

let errorCode = 'UNAUTHORIZED' 
console.log(i18next.t(`translation:errorCodes.${errorCode}`))
errorCode = 'BAD REQUEST'
console.log(i18next.t(`translation:errorCodes.${errorCode}`))
```

## Expected behavior

With version 20.6.1, the code above prints

```
Unauthorized
Bad request
```
With version 21.1.0, it prints

```
Unauthorized
translation:errorCodes.BAD REQUEST
```

## Your Environment

(Probably not relevant, but node v14.17.6 on MacOS)

**Comment by adrai** [comment]
should be fixed with v21.1.1

**Comment by pasieronen** [comment]
Thanks for fixing this so quickly!

One thing I noticed was that v20.x was less strict about mixing namespace and key separators, and our code had many places that (incorrectly) used just colons (e.g. `translations:something:errorCodes:UNAUTHORIZED`). That still works in v21.x, _except_ if combined with spaces - then it has to be `translations:something.errorCodes.BAD REQUEST`).
